File: app/src/main/java/com/example/tradeup/data/repository/AuthRepository.java
// File: src/main/java/com/example/tradeup/data/repository/AuthRepository.java
package com.example.tradeup.data.repository;

import androidx.annotation.NonNull;
import com.google.firebase.auth.FirebaseUser;
import java.util.concurrent.CompletableFuture;

public interface AuthRepository {
    FirebaseUser getCurrentUser();

    CompletableFuture<FirebaseUser> registerUser(String email, String password);

    CompletableFuture<FirebaseUser> loginUser(String email, String password);

    void logoutUser();

    CompletableFuture<FirebaseUser> loginWithGoogle(String idToken);

    CompletableFuture<Void> sendPasswordResetEmail(String email);

    CompletableFuture<Void> sendEmailVerification(@NonNull FirebaseUser user);

    CompletableFuture<Void> reauthenticateAndDeleteCurrentUser(String password);
}

File: app/src/main/java/com/example/tradeup/data/repository/AuthRepository.kt
package com.example.tradeup.data.repository

import com.google.firebase.auth.FirebaseUser

interface AuthRepository {
    fun getCurrentUser(): FirebaseUser?
    suspend fun registerUser(email: String, pass: String): Result<FirebaseUser>
    suspend fun loginUser(email: String, pass: String): Result<FirebaseUser>
    fun logoutUser()
    suspend fun sendPasswordResetEmail(email: String): Result<Unit>
    suspend fun sendEmailVerification(user: FirebaseUser): Result<Unit>
    // Thêm các hàm khác nếu cần, ví dụ: sendEmailVerification
}